src/commands/attack.js

```diff
@@ -4,47 +4,49 @@ import { random20, chooseWeighted } from '../utils/random.js';
 * @param {object} drop An item to be added to the player's inventory
 * @param {string} drop.mat The item's name
 */
-const addItem = (drop, inventory) => {
-  let result = inventory.find(loot => loot.mat === drop.mat);
+const addItem = (drop, ctx) => {
+  let result = ctx.inventory.find(loot => loot.mat === drop.mat);
   if (result) {
     result.quantity += 1;
   } else {
-    inventory.push({ mat: drop.mat, quantity: 1 });
+    ctx.inventory.push({ mat: drop.mat, quantity: 1 });
   }
 };
 
-export const attack = (player, enemy, inventory) => {
-  let enemyDamage = Math.floor((player.atk * random20()) - (enemy.def * random20()));
+export const attack = (ctx) => {
+  let enemyDamage = Math.floor((ctx.player.atk * random20()) - (ctx.enemy.def * random20()));
```

Comment:
Continuing with the destructuring idea, you will need the whole `ctx` object in order to assign `ctx.enemy = null` at the end, but you can extract the player and inventory and previous enemy for reading still like this:
```suggestion
export const attack = (ctx) => {
  const { player, inventory, enemy } = ctx;
  let enemyDamage = Math.floor((player.atk * random20()) - (enemy.def * random20()));
```